src/flagStore.ts

```diff
@@ -76,6 +80,21 @@ export class FlagStore {
 				}
 			}
 			this.storeUpdates.fire(true);
+			this.on('update', async (keys: string) => {
+				const flagKeys = Object.values(keys);
+				flagKeys.map(key => {
+					const curVars = this.flagMetadata[key].variationLength;
+					this.store.get(DATA_KIND, key, async (res: FlagConfiguration) => {
+						if (!res) {
+							return;
+						}
+						if (curVars !== res.variations.length) {
+							this.flagMetadata[key] = await this.api.getFeatureFlag(this.config.project, key, this.config.env);
+							this.storeUpdates.fire(true);
+						}
+					});
+				});
+			});
```

Comment:
What is the purpose of this new block? Were updates not being made when the number of variations changed before?

Reply:
as of right now(before PR this one is based one) every time there's a flag update we get the latest flag via API. Per our conversation I changed it so the control plane data only updates on the `refreshRate`.

What this changes is so that it only updates on `refreshRate` except when there's a variation change. Because variations span the global flag config and environment targeting where almost nothing else does.